Modules/Microsoft365DSC/DSCResources/MSFT_EXOAuthenticationPolicy/MSFT_EXOAuthenticationPolicy.psm1

```diff
@@ -0,0 +1,587 @@
+        [Switch]
+        $AllowBasicAuthReportingWebServices,
+
+        [Parameter()]
+        [Switch]
+        $AllowBasicAuthRpc,
+
+        [Parameter()]
+        [Switch]
+        $AllowBasicAuthSmtp,
+
+        [Parameter()]
+        [Switch]
+        $AllowBasicAuthWebServices,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [System.String]
+        $CertificatePath,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $CertificatePassword
+    )
+
+    Write-Verbose -Message "Setting Authentication Policy configuration for $Identity"
+
+    $currentAuthenticationPolicyConfig = Get-TargetResource @PSBoundParameters
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace "MSFT_", ""
+    $CommandName  = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $ConnectionMode = New-M365DSCConnection -Workload 'ExchangeOnline' `
+        -InboundParameters $PSBoundParameters
+
+    $NewAuthenticationPolicyParams = @{
+        AllowBasicAuthActiveSync            = $AllowBasicAuthActiveSync
+        AllowBasicAuthAutodiscover          = $AllowBasicAuthAutodiscover
+        AllowBasicAuthImap                  = $AllowBasicAuthImap
+        AllowBasicAuthMapi                  = $AllowBasicAuthMapi
+        AllowBasicAuthOfflineAddressBook    = $AllowBasicAuthOfflineAddressBook
+        AllowBasicAuthOutlookService        = $AllowBasicAuthOutlookService
+        AllowBasicAuthPop                   = $AllowBasicAuthPop
+        AllowBasicAuthPowerShell            = $AllowBasicAuthPowerShell
+        AllowBasicAuthReportingWebServices  = $AllowBasicAuthReportingWebServices
+        AllowBasicAuthRpc                   = $AllowBasicAuthRpc
+        AllowBasicAuthSmtp                  = $AllowBasicAuthSmtp
+        AllowBasicAuthWebServices           = $AllowBasicAuthWebServices
+    }
+
+    # CASE: Authentication Policy doesn't exist but should;
+    if ($Ensure -eq "Present" -and $currentAuthenticationPolicyConfig.Ensure -eq "Absent")
+    {
+        Write-Verbose -Message "Authentication Policy '$($Identity)' does not exist but it should. Create and configure it."
+        New-AuthenticationPolicy -Name $Identity @NewAuthenticationPolicyParams
```

Comment:
This should probably by piped to | Null at the end to prevent the output from showing the creation results. We want to keep the verbose as clean as possible.

Reply:
New and set commands now pipe to null.